src/util/file_system.cpp

```diff
@@ -230,6 +232,27 @@ void open_path(const std::string& path)
 #endif
 }
 
+std::string escape_url(const std::string& url)
+{
+#ifndef __EMSCRIPTEN__
+  std::string result = url;
+  CURL *curl = curl_easy_init();
```

Comment:
Maybe it would be better to not initialize curl context every time `escape_url` is called?

Reply:
How else would you do this?